aten/src/ATen/cuda/PinnedMemoryAllocator.cpp

```diff
@@ -18,4 +18,9 @@ void PinnedMemoryAllocator::deallocate(void* ptr) const {
   return state->cudaHostAllocator->free(nullptr, ptr);
 }
 
+PinnedMemoryAllocator* getPinnedMemoryAllocator() {
+  static PinnedMemoryAllocator r;
```

Comment:
We've gone back and forth on how to do static initialization a few times (CC @goldsborough).  I'd rather just have one pattern -- can we do the same thing as the "UndefinedTensor _singleton" here?  (Or change that one however you want and make this the same).

Reply:
What's wrong with this? For objects with non-trivial destructors, it's safer to to use this pattern.

Reply:
Also fine, I just think it's slightly preferable to have a common way of doing things that people can point to / copy from.